src/backend/src/utils/validation.utils.ts

```diff
@@ -115,6 +116,16 @@ export const workPackageProposedChangesValidators = [
   nonEmptyString(body('workPackageProposedChanges.deliverables.*'))
 ];
 
+export const validateProposedChangesFields = async (
```

Comment:
this is too complicated. It looks like you're trying to abstract checks to see whether an object is undefined, I was suggesting you make a helper method that takes in the ids we're checking and just houses the duplicate code (checking for links, checking for project manager ids, and checking for project lead ids)

Reply:
This helper method is the type of thing that sounds good in theory, but isn't really good in implementation (as you found out). If you ever run into a situation like this where something just isn't working (especially something like this that's unlike any other methods we have), it's probably because it's not the best solution. You should take a step back, try to think of a simpler approach, or ask someone else to put eyes on your code to make sure the idea is a viable one